File: files/CH-123.R
library(tidyverse)
library(readxl)

path = "files/CH-123 Custom Grouping.xlsx"
input = read_excel(path, range = "B2:C26")
test  = read_excel(path, range = "G2:H5")


result = input %>%
  mutate(Group = (row_number() - 1) %/% 10 + 1) %>%
  summarise(`Total Sales` = sum(Sales),
            .by = Group)

all.equal(result, test)
#> [1] TRUE

File: files/100-199/123/CH-123.py
import pandas as pd

path = "CH-123 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=24)
test = pd.read_excel(path, usecols="G:H", skiprows=1, nrows=3)

result = input.assign(Group=input.index // 10 + 1).drop(columns=["Date"]).groupby("Group").sum().rename(columns={"Sales": "Total Sales"}).reset_index()

print(result.equals(test))  # True